pkg/alertmanager/api.go

```diff
@@ -246,6 +245,7 @@ func validateUserConfig(logger log.Logger, cfg alertspb.AlertConfigDesc, limits
 	}
 	defer os.RemoveAll(userTempDir)
 
+	templateFiles := make([]string, 0, len(cfg.Templates))
```

Comment:
Not exactly, `amCfg.Templates` can also have glob-style fields like `*.tpl`. Also, `amCfg.Templates` is simply a list of names with no content, where `cfg.Templates` is a map of template name to template content.

Since this is about content validation, only `cfg.Templates` can even have content in the first place, so we don't need to consider `amCfg.Templates` at all.